src/mappings/talent-layer-review.ts

```diff
@@ -15,15 +15,30 @@ export function handleMint(event: Mint): void {
   review.createdAt = event.block.timestamp
   review.cid = event.params.reviewUri
 
-  let user = User.load(event.params.toId.toString())
-  if (!user) return
-  const rating = user.rating
-    .times(user.numReviews.toBigDecimal())
+  const receiver = User.load(event.params.toId.toString())
+  const receiverStats = getOrCreateUserStats(event.params.toId);
+
+  if (!receiver) return
+
+  receiver.rating
+    .times(receiverStats.numReceivedReviews.toBigDecimal())
     .plus(event.params.rating.toBigDecimal())
-    .div(user.numReviews.plus(ONE).toBigDecimal())
-  user.rating = rating
-  user.numReviews = user.numReviews.plus(ONE)
-  user.save()
+    .div(receiverStats.numReceivedReviews.plus(ONE).toBigDecimal())
+  receiverStats.numGivenReviews.plus(ONE)
+  receiver.save()
+  receiverStats.save()
+
+  const service = getOrCreateService(event.params.serviceId)
+  const buyerStats = getOrCreateUserStats(BigInt.fromString(service.buyer!))
```

Comment:
had to add "!" to "service.buyer!" & "service.seller!" for the compiler.

In theory these fields cannot be null at this stage, so should not be a risk to break the graph.